### PR TITLE
Lysenko/ocd 2980 create edit command

### PR DESCRIPTION
**Pull request by konstantin-lysenko-netapp**

### CODE REVIEW

File: cmd/apply.go [hunk 1]
```diff
@@ -36,32 +30,39 @@ and see the samples dir`
 			validateToken(context.Background())
 		},
 		PreRunE: func(cmd *cobra.Command, args []string) error {
-			return validateFlags(cmd, args)
+			return validateFlags()
 		},
 		Run: func(cmd *cobra.Command, args []string) {
-			runApplyCmd(context.Background())
+			configHandler, err := newConfigurationHandler(fileToApply)
```
Comment (caduri) [inline cmd/apply.go:36]: revert this code into runApplyCmd and do the same for edit and create

File: cmd/apply.go [hunk 2]
```diff
@@ -36,32 +30,39 @@ and see the samples dir`
...
-			fmt.Printf("Failed to apply resource - %s\n", err.Error())
-		}
+		return handleYamlFile, nil
+	default:
+		return nil, errors.New("wrong file extension: JSON and YAML formats are accepted only")
 	}
 }
 
-func handleYamlFile(ctx context.Context) error {
+type configHandler func(context.Context, commandHandler) error
```
Comment (caduri) [inline cmd/apply.go:61]: refactor the generic part that all three cmds (apply, create and edit) are using into different class

**Review by caduri — CHANGES_REQUESTED** [review]

File: pkg/utils/helpers.go [hunk 3]
```diff
@@ -196,3 +198,82 @@ func GetNounForm(noun string, length int) string {
...
+func NewConfigHandler(filepath string) (ConfigHandler, error) {
+	fileExtension := fp.Ext(filepath)[1:]
+
+	switch fileExtension {
+	case "json":
+		return &JsonConfigHandler{Config{Filename: filepath}}, nil
+	case "yaml", "yml":
+		return &YamlConfigHandler{Config{Filename: filepath}}, nil
+	default:
+		return nil, errors.New("wrong file extension: JSON and YAML formats are accepted only")
```
Comment (caduri) [inline pkg/utils/helpers.go:211]: I think the error should be "wrong file extension: Only Json and Yaml formats are supported"

File: pkg/utils/helpers.go [hunk 4]
```diff
@@ -159,7 +161,7 @@ func ConvertYamlFileToMap(fileName string) ([]map[string]interface{}, error) {
 	return retVal, nil
 }
 
-func ConvertYamlFileToArrayOfMaps(fileName string) ([]map[string]interface{}, error) {
+func (h *YamlConfigHandler) ToArrayOfMaps(fileName string) ([]map[string]interface{}, error) {
```
Comment (caduri) [inline pkg/utils/helpers.go:164]: rearranage this file to have the type next to its functions
or whatever order that you think its best
for readability puposes

**Review by caduri — CHANGES_REQUESTED** [review]

File: pkg/utils/helpers.go [hunk 5]
```diff
@@ -2,12 +2,14 @@ package utils
...
+	"context"
 	"encoding/json"
 	"errors"
 	"fmt"
 	"gopkg.in/yaml.v3"
 	"io"
 	"io/ioutil"
+	fp "path/filepath"
 	"spot-oceancd-cli/pkg/oceancd/model"
 	//"fmt"
```
Comment (caduri) [inline pkg/utils/helpers.go:14]: why do we have comments here?

File: pkg/utils/helpers.go [hunk 6]
```diff
@@ -25,83 +27,86 @@ var (
 	}
 )
 
-func GetOceanCdEntityKindByName(entityType string) (string, error) {
-	if entityType == "cluster" || entityType == "Cluster" || entityType == "clusters" || entityType == "Clusters" {
-		return "", errors.New(fmt.Sprintf("error: Unrecognize resource type '%s'", entityType))
-	}
-
-	return GetEntityKindByName(entityType)
+type ConfigHandler interface {
```
Comment (caduri) [inline pkg/utils/helpers.go:30]: I didn't see the size of this file and I think we should split it multiple helpers files by type
json_conffiguration_helpers
yaml_conffiguration_helpers
etc
wdyt?

**Review by caduri — CHANGES_REQUESTED** [review]

File: pkg/utils/helpers.go (hunk 5 above)
Comment (konstantin-lysenko-netapp) [reply pkg/utils/helpers.go:14]: it isn't my change, I dunno :)
wanna remove it?

File: pkg/utils/helpers.go (hunk 6 above)
Comment (konstantin-lysenko-netapp) [reply pkg/utils/helpers.go:30]: I guess the configuration_hendlers file will be enough
ok?

**Review by caduri — CHANGES_REQUESTED** [review]

**Review by caduri — APPROVED** [review]

**Pull request merged**